src/llvm/di.rs

```diff
@@ -20,6 +20,29 @@ use std::hash::Hasher;
 // backward compatibility
 const MAX_KSYM_NAME_LEN: usize = 128;
 
+pub struct DIType {
+    metadata: LLVMMetadataRef,
+}
+
+impl DIType {
+    pub fn new(metadata: LLVMMetadataRef) -> Self {
+        Self { metadata }
+    }
+
+    pub fn name(&self) -> Option<&CStr> {
+        let mut len = 0;
+        // `LLVMDITypeGetName` doesn't allocate any memory, it just returns
+        // an existing pointer:
+        // https://github.com/llvm/llvm-project/blob/eee1f7cef856241ad7d66b715c584d29b1c89ca9/llvm/lib/IR/DebugInfo.cpp#L1489-L1493
+        //
+        // Therefore, we don't need to call `LLVMDisposeMessage`. The memory
+        // gets freed when calling `LLVMDisposeDIBuilder`. Example:
+        // https://github.com/llvm/llvm-project/blob/eee1f7cef856241ad7d66b715c584d29b1c89ca9/llvm/tools/llvm-c-test/debuginfo.c#L249-L255
+        let ptr = unsafe { LLVMDITypeGetName(self.metadata, &mut len) };
+        (!ptr.is_null()).then(|| unsafe { CStr::from_ptr(ptr) })
```

Comment:
nit: this is ptr::NonNull::new(ptr).map(...)

Reply:
Alright. We use the `(!ptr.is_null()).then(|| ...)` way in other places too, I will make a separate PR to change them too.